mkfs/rootdir.c

```diff
@@ -471,8 +672,29 @@ static int add_file_items(struct btrfs_trans_handle *trans,
 			goto end;
 		}
 
-		ret = btrfs_insert_inline_extent(trans, root, objectid, 0,
-						 buffer, st->st_size);
+		switch (g_compression) {
+		case BTRFS_COMPRESS_ZLIB:
+			ret = zlib_compress_inline_extent(buffer, st->st_size,
+							  &comp_buf, &comp_size);
+			if (ret < 0)
+				goto end;
+			break;
+		default:
+			ret = 0;
+			break;
+		}
+
+		if (ret == 1) {
```

Comment:
Please do not use immediate number other than 0, so this should be `if (ret > 0) {`